kola/tests/misc/tang.go

```diff
@@ -0,0 +1,300 @@
+package misc
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+
+	tutil "github.com/flatcar/mantle/kola/tests/util"
+
+	"io/fs"
+
+	"github.com/anatol/tang.go"
+	"github.com/flatcar/mantle/kola/cluster"
+	"github.com/flatcar/mantle/kola/register"
+	"github.com/flatcar/mantle/kola/tests/util"
+	"github.com/flatcar/mantle/platform"
+	"github.com/flatcar/mantle/platform/conf"
+	"github.com/flatcar/mantle/platform/machine/qemu"
+	"github.com/flatcar/mantle/platform/machine/unprivqemu"
+)
+
+const (
+	// In the Ignition config below, we create a partition labeled ROOT on a second disk.
+	// As advised in the Flatcar documentation, we then remove the ROOT label from the existing
+	// root partition, which is vda9 in the QEMU disk image.
+	IgnitionConfigRootTang = `{
+		"ignition": {
+			"config": {},
+			"timeouts": {},
+			"version": "3.3.0"
+		},
+		"networkd": {},
+		"storage": {
+			"disks": [
+				{
+					"device": "/dev/disk/by-id/virtio-secondary",
+					"wipeTable": true,
+					"partitions": [
+						{
+							"label": "root",
+							"number": 1
+						}
+					]
+				}
+			],
+			"luks": [
+				{
+					"name": "rootencrypted",
+					"device": "/dev/disk/by-partlabel/root",
+					"clevis": {
+						"tang": [
+							{
+								"url": "http://{{ .TangIP }}:{{ .TangPort }}",
+								"thumbprint": "HkwVNDeKhzaVqWhXtXwEIGNILRZt4cBWWb0kI1-a0NM"
+							}
+						]
+					}
+				}
+			],
+			"filesystems": [
+				{
+					"device": "/dev/disk/by-id/dm-name-rootencrypted",
+					"format": "ext4",
+					"label": "ROOT"
+				}
+			]
+		},
+		"systemd": {
+			"units": [{
+			"name": "remove-root-label.service",
+			"enabled": true,
+			"contents": "[Service]\nType=oneshot\nExecStart=wipefs -a /dev/vda9\n[Install]\nWantedBy=multi-user.target"
```

Comment:
Isn't it possible to reformat this partition with Ignition? I would prefer that because otherwise it's not clear which rootfs is used in the initrd in the first run.

Reply:
I think some filesystems entry like that could work, here in butane yaml:

```
variant: flatcar
version: 1.0.0
storage:
  filesystems:
  - device: /dev/disk/by-partlabel/ROOT
    wipe_filesystem: true
    format: none
```
Ignition json:
```
{
  "ignition": {
    "version": "3.3.0"
  },
  "storage": {
    "filesystems": [
      {
        "device": "/dev/disk/by-partlabel/ROOT",
        "format": "none",
        "wipeFilesystem": true
      }
    ]
  }
}

```

Reply:
Yes, that works. I changed it as you suggested.